biz.aQute.bndlib/src/aQute/bnd/build/WorkspaceRepository.java

```diff
@@ -42,14 +42,15 @@ public WorkspaceRepository(Workspace workspace) {
 			if (Constants.VERSION_ATTR_LATEST.equals(range) || matchVersion(range, version, exact)) {
 				File file = project.getOutputFile(bsn, version.toString());
 				if (!file.exists()) {
-					Builder builder = project.getSubBuilder(bsn);
-					Jar jar = builder.build();
-					if (jar == null) {
-						project.getInfo(builder);
-						continue;
+					try (Builder builder = project.getSubBuilder(bsn);) {
+						Jar jar = builder.build();
```

Comment:
You can still use a try-with-resources even if `builder.build()` returns `null`. This would make things more consistent and you can remove the explicit call to `jar.close()`.